File: ServerFiles/bridge/resources/rageOS/Global/ServerTime.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Timers;
using GTANetworkAPI;

namespace RageOS
{
    public class ServerTime : Script
    {
        public static string[] dayNames = { "Montag", "Dienstag", "Mittwoch", "Donnerstag", "Freitag", "Samstag", "Sonntag" };
        System.Threading.Thread timeThread = null;
        private Timer timer;

        public ServerTime()
        {
            API.OnResourceStart += DayNightInit;
            API.OnResourceStop += DayNightExit;
        }

        public void DayNightInit()
        {
            API.Delay(5000, true, () =>
            {
                int day = (int)DateTime.Now.DayOfWeek;
                API.SetWorldSyncedData("DAYNIGHT_DAY", day); // aka dayNames[0] - Monday
                API.SetWorldSyncedData("DAYNIGHT_DAY_STRING", dayNames[day]);
                API.SetWorldSyncedData("DAYNIGHT_HOUR", DateTime.Now.Hour);
                API.SetWorldSyncedData("DAYNIGHT_MINUTE", DateTime.Now.Minute);
                API.SetWorldSyncedData("DAYNIGHT_RENDER_ICON", false);
                DayNightPrepareText();

                foreach (var player in API.GetAllPlayers()) API.FreezePlayerTime(player, true);

                timeThread = new System.Threading.Thread(UpdateTime);
                timeThread.Start();

                timer = new Timer();
                timer.Interval = 10000;
                timer.Elapsed += TimerEventHandler;
                timer.AutoReset = true;
                timer.Enabled = true;
            });
        }

        public void DayNightExit()
        {
            timer.Stop();
            if (timeThread != null) timeThread.Abort();
            timeThread = null;
        }

        public static void DayNightPrepareText()
        {
            API.Shared.SetWorldSyncedData("DAYNIGHT_TEXT", API.Shared.GetWorldSyncedData("DAYNIGHT_HOUR").ToString("D2") + ":" + API.Shared.GetWorldSyncedData("DAYNIGHT_MINUTE").ToString("D2") + " Uhr");
        }

        public void TimerEventHandler(System.Object source, ElapsedEventArgs e)
        {
            UpdateTime();
        }
        public void UpdateTime()
        {
            int add = 1;
            int minute = API.GetWorldSyncedData("DAYNIGHT_MINUTE");
            int hour = API.GetWorldSyncedData("DAYNIGHT_HOUR");
            if (hour >= 22 || hour <= 4)
            {
                add = 2;
            }
            minute = minute + add;

            if (minute >= 60)
            {
                minute = 0;
                hour = hour + 1;

                API.SetWorldSyncedData("DAYNIGHT_MINUTE", minute);

                if (hour >= 24)
                {
                    hour = 0;
                    API.SetWorldSyncedData("DAYNIGHT_HOUR", hour);

                    API.SetWorldSyncedData("DAYNIGHT_DAY", API.GetWorldSyncedData("DAYNIGHT_DAY") + 1);
                    if (API.GetWorldSyncedData("DAYNIGHT_DAY") == dayNames.Length) API.SetWorldSyncedData("DAYNIGHT_DAY", 0);
                    API.SetWorldSyncedData("DAYNIGHT_DAY_STRING", dayNames[API.GetWorldSyncedData("DAYNIGHT_DAY")]);
                }
                else
                {
                    API.SetWorldSyncedData("DAYNIGHT_HOUR", hour);
                }
            }
            else
            {
                API.SetWorldSyncedData("DAYNIGHT_MINUTE", minute);
            }

            API.SetTime(hour, minute, 0);
            DayNightPrepareText();
        }

    }
}

File: ServerFiles/bridge/resources/rageOS/Global/ServerTime.js
var oldDateTime = 0;
var player = null;
var PLAYER_NAME = "";
var DAYNIGHT_DAY_STRING = "";
var DAYNIGHT_TEXT = "";

API.onResourceStart.connect(function() {
	player = API.GetLocalPlayer();
});

API.onUpdate.connect(function () {
    if (!API.GetHudVisible()) return;
	var dateTime = API.GetGameTime();
	if (dateTime - oldDateTime >= 2000)
	{
		player = API.GetLocalPlayer();
		oldDateTime = dateTime;
		PLAYER_NAME = API.GetEntitySyncedData(player, "PLAYER_NAME");
		DAYNIGHT_DAY_STRING = API.GetWorldSyncedData("DAYNIGHT_DAY_STRING");
		DAYNIGHT_TEXT = API.GetWorldSyncedData("DAYNIGHT_TEXT");
	}

    var res = API.GetScreenResolutionMaintainRatio();

	if(API.hasEntitySyncedData(player, "HUD") && API.GetEntitySyncedData(player, "HUD") == true)
	{
		API.drawText(PLAYER_NAME, (res.Width / 2) - 600, res.Height - 145, 0.65, 255, 255, 255, 255, 4, 0, true, true, 0);
		API.drawText(DAYNIGHT_DAY_STRING, (res.Width / 2) - 600, res.Height - 105, 0.40, 241, 196, 15, 255, 4, 0, true, true, 0);
		API.drawText(DAYNIGHT_TEXT, (res.Width / 2) - 600, res.Height - 80, 0.65, 255, 255, 255, 255, 4, 0, true, true, 0);
	}
});
